src/core/sock.c

```diff
@@ -658,6 +658,12 @@ int nn_sock_recv (struct nn_sock *self, struct nn_msg *msg, int flags)
             return rc;
         }
 
+		/* Have to check if one broken connection needs a reconnection at all cases */
+		if(nn_fast (rc == -EAGAIN) && nn_fast(self->statistics.broken_connections > 0)) {
```

Comment:
nn_fast feels wrong here.  For two reasons:

1. We're already in an error path -- this should never be hot -- feels like premature optimization
2. Is it the case that when in this error path, this is really the most  likely case?

There are indent and spacing issues as Dirkjan already noted, as well.